keras_cv/visualization/plot_bounding_box_gallery.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import functools
+
+import numpy as np
+import tensorflow as tf
+
+import keras_cv
+import keras_cv.visualization.draw_bounding_boxes
+from keras_cv import utils
+from keras_cv.utils import assert_cv2_installed
+from keras_cv.utils import assert_matplotlib_installed
+from keras_cv.visualization.plot_gallery import plot_gallery
+
+try:
+    from matplotlib import patches
+except:
+    patches = None
+
+
+def plot_bounding_box_gallery(
+    images,
+    value_range,
+    bounding_box_format,
+    pred_color=(255, 235, 59),
+    true_color=(0, 188, 212),
+    y_true=None,
+    y_pred=None,
+    thickness=2,
+    font_scale=1.0,
+    text_thickness=None,
+    class_mapping=None,
+    ground_truth_mapping=None,
+    prediction_mapping=None,
+    legend=False,
+    legend_handles=None,
+    rows=3,
+    cols=3,
+    **kwargs
+):
+    """plots a gallery of images with corresponding bounding box annotations
+
+    ![Example bounding box gallery](https://i.imgur.com/Fy7kMnP.png)
+
+    Args:
+        images: a Tensor or NumPy array containing images to show in the gallery.
+        value_range: value range of the images.
+        bounding_box_format: the bounding_box_format  the provided bounding boxes are
+            in.
+        y_true: a KerasCV bounding box dictionary representing the ground truth bounding
+            boxes.
+        y_pred: a KerasCV bounding box dictionary representing the predicted truth
+            bounding boxes.
+        pred_color: three element tuple representing the color to use for plotting
+            predicted bounding boxes.
+        true_color: three element tuple representing the color to use for plotting
```

Comment:
for ground truth, I usually use white with 50% transparency to make them visually distinct from predictions which use the high contrast lines.